### PR TITLE
Remove defaultProps from SegmentedControlIOS

### PR DESCRIPTION
**Pull request by joxw1**
## Summary

Issue #31604 . Remove `defaultProps` from `SegmentedControlIOS`.

## Changelog

[JavaScript] [Changed] - Remove defaultProps from SegmentedControlIOS

## Test Plan
Added tests for `SegmentedControlIOS` pass.

### CODE REVIEW

File: Libraries/Components/SegmentedControlIOS/SegmentedControlIOS.ios.js [hunk 1]
```diff
@@ -111,7 +106,14 @@ const SegmentedControlIOSWithRef = React.forwardRef(
     props: SegmentedControlIOSProps,
     forwardedRef: ?React.Ref<typeof RCTSegmentedControlNativeComponent>,
   ) => {
-    return <SegmentedControlIOS {...props} forwardedRef={forwardedRef} />;
+    return (
+      <SegmentedControlIOS
+        {...props}
+        enabled={props.enabled === undefined ? true : props.enabled}
```
Comment (lunaleaps) [inline Libraries/Components/SegmentedControlIOS/SegmentedControlIOS.ios.js:112]: I think to be consistent with us replacing `defaultProps`, I'd lean towards doing this in `SegmentedControlIOS`. 

If you do that, you can also pull the props out on line 86 and use the default assignment `const {values = [], ....} = this.props`

**Review by lunaleaps — CHANGES_REQUESTED** [review]

File: Libraries/Components/SegmentedControlIOS/__tests__/SegmentedContolIOS-test.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+/**
```
Comment (lunaleaps) [inline Libraries/Components/SegmentedControlIOS/__tests__/SegmentedContolIOS-test.js:1]: Amazing! Thanks for writing these tests!

Comment (joxw1) [reply Libraries/Components/SegmentedControlIOS/__tests__/SegmentedContolIOS-test.js:1]: You're welcome! Thanks for your reviews and your patience with me!

**Review by lunaleaps — APPROVED** [review]

**Comment by lunaleaps** [comment]
Thanks @joxw1 for your help!! I modified slightly the prop values from an internal review to make things simpler since we're only using these props in one place. Thank you again!

**Pull request closed without merging**